File: frontend/src/types/order.ts
/**
 * Order Type Definitions
 *
 * Defines types for orders, line items, and order status tracking.
 * Based on specs/002-cart-order-management/data-model.md
 */

/**
 * Order status type union
 * Represents the lifecycle stage of an order from cart to completion
 */
export type OrderStatus = 'new' | 'submitted' | 'paid' | 'completed' | 'cancelled' | 'refunded' | 'on-hold' | 'processing';

/**
 * Order entity
 * Represents both shopping carts (status='new') and submitted orders (status!='new')
 * Orders are market-specific
 */
export interface Order {
  id: string;
  tenantId: string;
  marketId: string; // NEW: Orders are market-specific
  customerId: string;
  orderNumber: string;
  status: OrderStatus;
  subtotal: number;
  tax: number;
  shipping: number;
  discount: number;
  total: number;
  currency: string;
  isGuest: boolean;
  guestEmail?: string;
  createdAt: string; // ISO 8601
  updatedAt: string; // ISO 8601
  submittedAt?: string; // ISO 8601
  completedAt?: string; // ISO 8601
  cancelledAt?: string; // ISO 8601
  cancellationReason?: string;
  metadata?: Record<string, any>;

  // Relationships (populated in API responses)
  lineItems?: OrderLineItem[];
  shippingAddress?: any; // ShippingAddress from address.ts
  billingAddress?: any; // BillingAddress from address.ts
  transactions?: any[]; // PaymentTransaction from payment.ts
  statusHistory?: OrderStatusHistory[];
}

/**
 * Order line item entity
 * Represents a single product entry within an order
 */
export interface OrderLineItem {
  id: string;
  orderId: string;
  productId: string;
  marketId: string; // NEW: Product's market (should match order's market)
  productName: string;
  sku: string;
  productImage?: string;
  quantity: number;
  unitPrice: number;
  lineTotal: number;
  currency: string;
  stockQuantity?: number; // Current stock level for out-of-stock warnings
  metadata?: Record<string, any>;
}

/**
 * Order status history entry
 * Audit trail of all status changes for an order
 */
export interface OrderStatusHistory {
  id: string;
  orderId: string;
  oldStatus?: OrderStatus;
  newStatus: OrderStatus;
  changedAt: string; // ISO 8601
  changedBy?: string;
  notes?: string;
  metadata?: Record<string, any>;
}

File: mock-api/MockApi/Models/Order.cs
using System.Text.Json.Serialization;

namespace MockApi.Models;

public class Order
{
    public string Id { get; set; } = string.Empty;
    public string TenantId { get; set; } = string.Empty;
    public string MarketId { get; set; } = string.Empty;
    public string OrderNumber { get; set; } = string.Empty;
    public string Status { get; set; } = "pending";
    public decimal Subtotal { get; set; }
    public decimal Tax { get; set; }

    [JsonPropertyName("shipping")]
    public decimal ShippingCost { get; set; }
    public decimal Total { get; set; }
    public CustomerInfo Customer { get; set; } = new();
    public Address ShippingAddress { get; set; } = new();
    public Address? BillingAddress { get; set; }

    [JsonPropertyName("lineItems")]
    public List<OrderItem> Items { get; set; } = new();

    public string? TrackingNumber { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }
}

public class OrderItem
{
    public string Id { get; set; } = string.Empty;
    public string ProductId { get; set; } = string.Empty;
    public string ProductName { get; set; } = string.Empty;
    public string Sku { get; set; } = string.Empty;
    public string? ProductImageUrl { get; set; }
    public decimal UnitPrice { get; set; }
    public int Quantity { get; set; }

    [JsonPropertyName("lineTotal")]
    public decimal Subtotal { get; set; }

    public string Currency { get; set; } = "USD";
}

public class CustomerInfo
{
    public string? CustomerId { get; set; }
    public string FullName { get; set; } = string.Empty;
    public string Email { get; set; } = string.Empty;
    public string? Phone { get; set; }
}

public class Address
{
    public string Street { get; set; } = string.Empty;
    public string? Street2 { get; set; }
    public string City { get; set; } = string.Empty;
    public string State { get; set; } = string.Empty;
    public string PostalCode { get; set; } = string.Empty;
    public string Country { get; set; } = string.Empty;
}

public class CreateOrderRequest
{
    public string SessionId { get; set; } = string.Empty;
    public CustomerInfo Customer { get; set; } = new();
    public Address ShippingAddress { get; set; } = new();
    public Address? BillingAddress { get; set; }
    public List<OrderItemRequest> Items { get; set; } = new();
}

public class OrderItemRequest
{
    public string ProductId { get; set; } = string.Empty;
    public int Quantity { get; set; }
    public decimal UnitPrice { get; set; }
}

public class UpdateOrderStatusRequest
{
    public string Status { get; set; } = string.Empty;
    public string? Notes { get; set; }
}
